src/Ombi.TheMovieDbApi/TheMovieDbApi.cs

```diff
@@ -99,65 +105,95 @@ public async Task<List<MovieSearchResult>> SimilarMovies(int movieId, string lan
         }
 
-        public async Task<List<MovieSearchResult>> PopularMovies(string langageCode)
+        public async Task<List<MovieSearchResult>> PopularMovies(string langCode)
         {
-            var request = new Request($"movie/popular", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+            request.AddQueryString("without_keywords", settings.ExcludedKeywordIds);
+
             AddRetry(request);
             var result = await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> TopRated(string langageCode)
+        public async Task<List<MovieSearchResult>> TopRated(string langCode)
         {
-            var request = new Request($"movie/top_rated", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("vote_count.gte", "250");
+            request.AddQueryString("sort_by", "vote_average.desc");
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+            request.AddQueryString("without_keywords", settings.ExcludedKeywordIds);
+
             AddRetry(request);
             var result = await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> Upcoming(string langageCode)
+        public async Task<List<MovieSearchResult>> Upcoming(string langCode)
         {
-            var request = new Request($"movie/upcoming", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("with_release_type", "2|3");
```

Comment:
Why do we need to specify the release type?

Reply:
@tidusjar To achieve parity with `movies/upcoming` since [the docs say](https://developers.themoviedb.org/3/movies/get-upcoming) that endpoint uses these release types.